lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1467,93 +1407,331 @@ llvm::Expected<CompilerType> SwiftLanguageRuntime::GetChildCompilerTypeAtIndex(
+      auto get_name = [&]() {
+        std::string child_name;
+        llvm::raw_string_ostream(child_name) << idx;
+        return child_name;
+      };
+      auto get_info = [&]() -> llvm::Expected<ChildInfo> {
+        ChildInfo child;
+        child.byte_size = el_ti->getSize();
+        child.byte_offset = el_ti->getStride() * idx;
+        // FIXME:
+        // if (!ignore_array_bounds &&
+        //     (int64_t)child_byte_offset > (int64_t)ati->getSize())
+        //   return llvm::createStringError("array index out of bounds");
+
+        child.bitfield_bit_size = 0;
+        child.bitfield_bit_offset = 0;
+        child.is_base_class = false;
+        child.is_deref_of_parent = false;
+        child.language_flags = 0;
+        return child;
+      };
+      return visit_callback(type, 0, get_name, get_info);
+    };
+    for (unsigned i = 0; i < count; ++i)
+      if (!visit_only || *visit_only == i)
+        if (auto err = visit_element(i))
+          return err;
+    return success;
   }
   if (llvm::dyn_cast_or_null<swift::reflection::BuiltinTypeInfo>(ti)) {
     // Clang enums have an artificial rawValue property. We could
     // consider handling them here, but
     // TypeSystemSwiftTypeRef::GetChildCompilerTypeAtIndex can also
     // handle them and without a Process.
-    return CompilerType();
+    if (!TypeSystemSwiftTypeRef::IsBuiltinType(m_type)) {
+      LLDB_LOG(GetLog(LLDBLog::Types),
+               "{0}: unrecognized builtin type info or this is a Clang type "
+               "without DWARF debug info",
+               m_type.GetMangledTypeName());
+      return llvm::createStringError("missing debug info for Clang type \"" +
+                                     m_type.GetDisplayTypeName().GetString() +
+                                     "\"");
+    }
+    if (count_only)
+      return 0;
+    if (auto err = visit_callback(
+            CompilerType(), 0, []() { return ""; },
+            []() { return ChildInfo(); }))
+      return err;
+    return success;
   }
-  LogUnimplementedTypeKind(__FUNCTION__, type);
+  LogUnimplementedTypeKind(__FUNCTION__, m_type);
   return llvm::createStringError("not implemented");
 }
 
+  
+llvm::Expected<uint32_t>
+SwiftLanguageRuntime::GetNumFields(CompilerType type,
+                                   ExecutionContext *exe_ctx) {
+  if (exe_ctx)
+    return GetNumChildren(type, exe_ctx->GetBestExecutionContextScope(), false,
+                          false);
+ return llvm::createStringError("no execution context");
+}
+
+llvm::Expected<uint32_t> SwiftLanguageRuntime::GetNumChildren(
+    CompilerType type, ExecutionContextScope *exe_scope,
+    bool include_superclass, bool include_clang_types) {
+  LLDB_SCOPED_TIMER();
+  SwiftRuntimeTypeVisitor visitor(*this, type, exe_scope, !include_superclass,
+                                  include_clang_types);
+  return visitor.CountChildren();
+}
+
+static std::pair<SwiftLanguageRuntime::LookupResult, std::optional<size_t>>
```

Comment:
Could you add a doxygen comment on what the return pair means?

Reply:
Even better! I deleted the function because it is now dead code.